apps/meteor/client/lib/chats/flows/processMessageUploads.ts

```diff
@@ -0,0 +1,157 @@
+			}
+			return null;
+		}),
+	);
+
+	for (let i = 0; i < filesToUpload.length; i++) {
+		const attachment: FileAttachmentProps = {
+			title: filesToUpload[i].file.name,
+			type: 'file',
+			title_link: filesToUpload[i].url,
+			title_link_download: true,
+			encryption: {
+				key: filesToUpload[i].encryptedFile.key,
+				iv: filesToUpload[i].encryptedFile.iv,
+			},
+			hashes: {
+				sha256: filesToUpload[i].encryptedFile.hash,
+			},
+		};
+
+		if (/^image\/.+/.test(filesToUpload[i].file.type)) {
+			const dimensions = imgDimensions[i];
+			attachments.push({
+				...attachment,
+				image_url: filesToUpload[i].url,
+				image_type: filesToUpload[i].file.type,
+				image_size: filesToUpload[i].file.size,
+				...(dimensions && {
+					image_dimensions: dimensions,
+				}),
+			});
+		} else if (/^audio\/.+/.test(filesToUpload[i].file.type)) {
+			attachments.push({
+				...attachment,
+				audio_url: filesToUpload[i].url,
+				audio_type: filesToUpload[i].file.type,
+				audio_size: filesToUpload[i].file.size,
+			});
+		} else if (/^video\/.+/.test(filesToUpload[i].file.type)) {
+			attachments.push({
+				...attachment,
+				video_url: filesToUpload[i].url,
+				video_type: filesToUpload[i].file.type,
+				video_size: filesToUpload[i].file.size,
+			});
+		} else {
+			attachments.push({
+				...attachment,
+				size: filesToUpload[i].file.size,
+				format: getFileExtension(filesToUpload[i].file.name),
+			});
+		}
+
+		const files = {
+			_id: filesToUpload[i].id,
+			name: filesToUpload[i].file.name,
+			type: filesToUpload[i].file.type,
+			size: filesToUpload[i].file.size,
+			format: getFileExtension(filesToUpload[i].file.name),
+		};
+
+		arrayOfFiles.push(files);
+	}
+
+	return e2eRoom.encryptMessageContent({
+		attachments,
+		files: arrayOfFiles,
+		file: arrayOfFiles[0],
+		msg,
+	});
+};
+
+export const processMessageUploads = async (chat: ChatAPI, message: IMessage) => {
+	const { tmid, msg } = message;
+	const room = await chat.data.getRoom();
+	const e2eRoom = await e2e.getInstanceByRoomId(room._id);
+
+	const store = tmid ? chat.threadUploads : chat.uploads;
+	const filesToUpload = store.get();
+
+	if (filesToUpload.length === 0) {
+		return false;
+	}
+
+	const { fileUrls, fileIds } = filesToUpload.reduce<{ fileUrls: string[]; fileIds: string[] }>(
+		(acc, upload) => {
+			if (!upload.url || !upload.id) {
+				return acc;
+			}
+
+			acc.fileIds.push(upload.id);
+			acc.fileUrls.push(upload.url);
+
+			return acc;
+		},
+		{ fileUrls: [], fileIds: [] },
+	);
+
+	const shouldConvertSentMessages = await e2eRoom?.shouldConvertSentMessages({ msg });
+
+	let content;
+	if (e2eRoom && shouldConvertSentMessages) {
+		content = await getEncryptedContent(filesToUpload as EncryptedUpload[], e2eRoom, msg);
+	}
+
+	const composedMessage: AtLeast<IMessage, 'msg' | '_id' | 'rid'> = {
+		...message,
+		tmid,
+		msg,
+		content,
+		...(e2eRoom && {
+			t: 'e2e',
```

Comment:
```suggestion
			t: 'e2e',
			msg: '',
```
e2e messages should not send the message content to the server.

Reply:
can u test again?

Reply:
Still happening